### ISSUE TITLE
Support `**` in parquet_scan

### ISSUE THREAD

**Issue by mauropagano** [duckdb/duckdb#5897]
### What happens?

There is currently no way to crawl a directory in `parquet_scan()`.
Even passing a list of globs would not work since an `IO Error` is raised if any of the glob fails to find a file.

```
select * from parquet_scan(['/tmp/a/*.parquet', '/tmp/a/*/*.parquet']);
Error: IO Error: No files found that match the pattern "/tmp/a/*/*.parquet"
```

Filing as per conversation on Discord

### To Reproduce

The code fails even before getting into pq code so any file would do it
```
mkdir /tmp/a
mkdir /tmp/a/b
touch /tmp/a/file1.parquet
D select * from parquet_scan(['/tmp/a/*.parquet', '/tmp/a/*/*.parquet']);
```

### OS:

Linux

### DuckDB Version:

0.6.0

### DuckDB Client:

0.6.0

### Full Name:

Mauro Pagano

### Affiliation:

Independent consultant

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
@lverdoes is working on this

**Comment by Mytherin** [comment]
Implemented now by #6627
